### ISSUE TITLE
the triplet loss is always equals to margin

### ISSUE THREAD

**Issue by whitewrist** [KaiyangZhou/deep-person-reid#240]
![1571289649(1)](https://user-images.githubusercontent.com/32037129/66979676-fc6f3d80-f0e0-11e9-817b-b2e0228f3a67.jpg)

**Comment by whitewrist** [comment]
This is my first time to train a reid model with triplet loss. Could you give me some hints? Thanks.

**Comment by whitewrist** [comment]
here are my key config:  
     
train:
  optim: 'amsgrad'
  lr: 0.0015
  max_epoch: 150
  batch_size: 128
  fixbase_epoch: 0
  open_layers: ['classifier']
  lr_scheduler: 'single_step'
  stepsize: [60]

loss:
  name: 'triplet'

sampler:
    train_sampler: 'RandomIdentitySampler'

data:
  type: 'image'
  sources: ['msmt17']
  targets: ['dukemtmcreid']
  height: 256
  width: 128
  combineall: True
  transforms: ['random_flip', 'color_jitter']

**Comment by whitewrist** [comment]
model:
  name: 'resnet50'
  pretrained: True

data:
  type: 'image'
  sources: ['msmt17']
  targets: ['dukemtmcreid']
  height: 256
  width: 128
  combineall: True
  transforms: ['random_flip', 'color_jitter']
  save_dir: 'log/resnet'

loss:
  name: 'triplet'
  softmax:
    label_smooth: True

train:
  optim: 'amsgrad'
  lr: 0.0015
  max_epoch: 150
  batch_size: 128
  fixbase_epoch: 0
  open_layers: ['classifier']
  lr_scheduler: 'single_step'
  stepsize: [60]

test:
  batch_size: 32
  dist_metric: 'euclidean'
  normalize_feature: False
  evaluate: False
  eval_freq: 10
  rerank: False
  visactmap: False

sampler:
    train_sampler: 'RandomIdentitySampler'

I think it may be my lr is too large,  and when I set it to 2e-4, the problem was fixed.

**Comment by whitewrist** [comment]
Thanks you, this repository really help me a lot!
